packages/fxa-auth-server/lib/senders/emails/fluentLocalizer.ts

```diff
@@ -0,0 +1,91 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { DOMLocalization } from '@fluent/dom';
+import { FluentBundle, FluentResource } from '@fluent/bundle';
+import { negotiateLanguages } from '@fluent/langneg';
+import { JSDOM } from 'jsdom';
+import * as fs from 'fs';
+import path from 'path';
+import { renderWithOptionalLayout, context } from './renderer';
+import { mergeAllFtlFiles } from './mergeFtlFiles';
+import availableLocales from 'fxa-shared/l10n/supportedLanguages.json';
+
+const OTHER_EN_LOCALES = ['en-NZ', 'en-SG', 'en-MY'];
+const baseDir = path.join(__dirname);
+
+class FluentLocalizer {
+  constructor() {}
+
+  async localizeEmail(
+    templateName: string,
+    layoutName: string,
+    mailSubject: string,
+    variables: Record<any, any>,
+    acceptLanguage: string
+  ) {
+    const htmlDocument = renderWithOptionalLayout(
+      templateName,
+      { ...variables, ...context },
+      layoutName
+    );
+
+    const { document } = new JSDOM(htmlDocument).window;
+
+    const userLocales: Array<string> = [];
+
+    const parseLocales = acceptLanguage.split(',');
+    for (let locale of parseLocales) {
+      locale = locale.replace(/^\s*/, '').replace(/\s*$/, '');
+      const [lang, qual] = locale.split(';');
+      userLocales.push(lang);
+    }
+
+    const currentLocales = negotiateLanguages(
+      [...userLocales],
+      [...OTHER_EN_LOCALES, ...availableLocales],
+      {
+        defaultLocale: 'en-US',
+      }
+    );
+
+    async function fetchResource(locale: string) {
```

Comment:
I believe this will attempt to fetch and merge ftl files on every attempt to localize an email. We could store this value in a Map and have an initalize function to do the work. Similar to https://github.com/mozilla/fxa/blob/5f63d848b9bbed1a7f478b671a018a6557d13b75/packages/fxa-auth-server/lib/senders/templates/index.js#L56

Reply:
Ohh yes, good point!